### PR TITLE
Fix `Package` task

### LINKED ISSUE #2536: PowerShell extension - Not signed PowerShell-Files
Hello,
in my company only signed PowerShell-Files are allowed to use.
Since that I wrote a script which checks all files if they are signed or not.

**In your PowerShell-Extension for VSCode nearly every file is signed but the list below** (please sign those files and update the plugin).
If you need I can upload the simple script to check all PowerShell-Files for future extension releases.

_Maybe thinking about to not use PowerShell files at all in the plugin and using an other language for it instead?_

```
---------------------------------------------------
NOT SIGNED - FILES
---------------------------------------------------

Filepath:  .\ms-vscode.powershell-2020.3.0
Filename: InvokePesterStub.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: SampleModule.psd1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: PSScriptAnalyzerSettings.psd1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: PromptExamples.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: StopTest.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: Stop-Process2.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: SampleModule.psm1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: PathProcessingWildcards.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: DebugTest.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: ContentViewTest.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: Build.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: PathProcessingNoWildcards.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: PathProcessingNonExistingPaths.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples
Filename: ExtensionExamples.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\examples\Tests
Filename: PathProcessing.Tests.ps1


Filepath: .\ms-vscode.powershell-2020.3.0\examples\Tests
Filename: SampleModule.Tests.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\modules\Plaster\1.1.3\Templates\AddPSScriptAnalyzerSettings
Filename: PSScriptAnalyzerSettings.psd1


Filepath:  .\ms-vscode.powershell-2020.3.0\modules\Plaster\1.1.3\Templates\NewPowerShellScriptModule
Filename: Module.psm1


Filepath:  .\ms-vscode.powershell-2020.3.0\modules\Plaster\1.1.3\Templates\NewPowerShellScriptModule\test
Filename: Module.T.ps1


Filepath:  .\ms-vscode.powershell-2020.3.0\modules\PowerShellEditorServices\Commands\Public
Filename: Clear-Host.ps1
```

I hope this list will help you.

### LINKED ISSUE #3293: Name built VSIX with version instead of `-insiders`
Pulled out of #3268 because it requires changing the build script.

### PR DESCRIPTION
**Pull request by andschwa**
This now leaves the VSIX named as `vsce` creates it:
`powershell-YYYY.MM.X.vsix`, instead of renaming it `PowerShell-Insiders.vsix`.
I can't think of anything this would break, since the upload is not actually
done with the deleted `Upload` task but with an ADO pipeline task.

This also fixes the path to `InvokePesterStub.ps1` which is now part of the
PowerShellEditorServices repository. This was necessary in order to sign it, as
it must be signed _before_ the VSIX is created. Moving it to the other repo was
the best way to solve the "chicken-egg" problem that presented.

Resolves #3293 and combined with https://github.com/PowerShell/PowerShellEditorServices/pull/1513, resolves #2536.

### CODE REVIEW

**Comment by andschwa** [comment]
@rjmholt Running a "release" to see if everything still works.

File: vscode-powershell.build.ps1 [hunk 1]
```diff
@@ -36,14 +36,6 @@ task ResolveEditorServicesPath -Before CleanEditorServices, BuildEditorServices,
     }
 }
 
-task UploadArtifacts {
```
Comment (andschwa) [inline vscode-powershell.build.ps1:39]: Unneeded. The artifact upload is done by a task.

File: vscode-powershell.build.ps1 [hunk 2]
```diff
@@ -145,15 +137,15 @@ task Package UpdateReadme, {
         throw "Unable to find PowerShell EditorServices"
     }
 
-    Write-Host "`n### Packaging PowerShell-insiders.vsix`n" -ForegroundColor Green
+    $packageName = "$($script:PackageJson.name)-$($script:PackageJson.version).vsix"
```
Comment (andschwa) [inline vscode-powershell.build.ps1:140]: @rjmholt Are you ok with `PowerShell` changing to `powershell`? This _is_ the name in the `package.json`.

File: vscode-powershell.build.ps1 [hunk 3]
```diff
@@ -145,15 +137,15 @@ task Package UpdateReadme, {
...
+    Write-Host "`n### Packaging $packageName`n" -ForegroundColor Green
     exec { & node ./node_modules/vsce/out/vsce package --no-gitHubIssueLinking }
 
-    # Change the package to have a static name for automation purposes
```
Comment (andschwa) [inline vscode-powershell.build.ps1:151]: @rjmholt You mentioned a while ago that there were a number of "automation purposes" for which this was named `PowerShell-Insiders.vsix`, but at this point, I don't think they're around any more. We've essentially rewritten the whole of the release process (and it's getting to be almost buttery smooth now). Am I missing anything?

**Comment by andschwa** [comment]
Oof, ok so this doesn't quite work. It's a chicken-egg problem. We need `InvokePesterStub.ps1` signed _before_ the build happens, so that it can go into the resulting VSIX, which itself then needs signing. Grr.

File: src/features/PesterTests.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export class PesterTestsFeature implements vscode.Disposable {
     private invokePesterStubScriptPath: string;
 
     constructor(private sessionManager: SessionManager) {
-        this.invokePesterStubScriptPath = path.resolve(__dirname, "../../../InvokePesterStub.ps1");
+        this.invokePesterStubScriptPath = path.resolve(__dirname, "../../../modules/PowerShellEditorServices/InvokePesterStub.ps1");
```
Comment (andschwa) [inline src/features/PesterTests.ts:21]: I will need to run the tests internally so I can merge https://github.com/PowerShell/PowerShellEditorServices/pull/1513 first.

Comment (andschwa) [reply src/features/PesterTests.ts:21]: It bothers me that the tests pass, because they shouldn't without that PR being merged. 😢

Comment (andschwa) [reply src/features/PesterTests.ts:21]: Works!

**Comment by andschwa** [comment]
```powershell
@andys-mac-mini ~/src/vscode-powershell andschwa/package-fix ≡
> . '/Users/andschwa/.vscode/extensions/ms-vscode.powershell-2021.6.2/modules/PowerShellEditorServices/InvokePesterStub.ps1' -ScriptPath '/Users/andschwa/src/PowerShellEditorServices/test/PowerShellEditorServices.Test.Shared/Symbols/PesterFile.tests.ps1' -All -Output 'FromPreference'

Importing Pester module...
WARNING: Failed to import Pester. You must install Pester module to run or debug Pester tests.
WARNING: You can install Pester by executing: Install-Module Pester -Scope CurrentUser -Force
PS> Install-Module Pester -Scope CurrentUser -Force

@andys-mac-mini ...itorServices.Test.Shared/Symbols andschwa/import-pester-stub ≡                                                            > . '/Users/andschwa/.vscode/extensions/ms-vscode.powershell-2021.6.2/modules/PowerShellEditorServices/InvokePesterStub.ps1' -ScriptPath '/Users/andschwa/src/PowerShellEditorServices/test/PowerShellEditorServices.Test.Shared/Symbols/PesterFile.tests.ps1' -All -Output 'FromPreference'                                                                                                                                                                                                                                                                                        
Importing Pester module...

Starting discovery in 1 files.
Discovery finished in 188ms.
Running tests.
[+] /Users/andschwa/src/PowerShellEditorServices/test/PowerShellEditorServices.Test.Shared/Symbols/PesterFile.tests.ps1 507ms (33ms|317ms)
Tests completed in 528ms
Tests Passed: 3, Failed: 0, Skipped: 0 NotRun: 0
PS> 
```

File: vscode-powershell.build.ps1 (hunk 3 above)
Comment (rjmholt) [reply vscode-powershell.build.ps1:151]: AFAICR I think we should be good

**Review by rjmholt — APPROVED** [review]

**Pull request merged**